tests/rptest/services/openmessaging_benchmark.py

```diff
@@ -363,10 +366,89 @@ def check_succeed(self, validate_metrics=True):
 
         self._metrics = metrics
 
-        if validate_metrics:
+        if validate_metrics and raise_exceptions:
             OMBSampleConfigurations.validate_metrics(self._metrics,
                                                      self.validator)
 
+        if validate_metrics and not raise_exceptions:
+            is_valid, results = OMBSampleConfigurations.validate_metrics(
+                self._metrics, self.validator, raise_exceptions=False)
+            #if not raise_exceptions:
+            return is_valid, results
+
+    def detect_spikes_by_percentile(self,
+                                    results,
+                                    significant_factor=1.5,
+                                    max_spike_width=1,
+                                    expected_max_latencies=None):
+        def detect_spikes_in_series(latency_series, expected_max):
+            # Sort the series to calculate the median value
+            sorted_series = sorted(latency_series)
+            median_value = sorted_series[len(sorted_series) // 2]
+            # Determine the upper bound for spike detection
+            upper_bound = min(median_value * significant_factor, expected_max)
+
+            high_latency_start = None
+            consecutive_high_latency_count = 0
+            isolated_spikes = []
+
+            for i, value in enumerate(latency_series):
+                if value > upper_bound:
+                    if high_latency_start is None:
+                        high_latency_start = i
+                        self.logger.debug(
+                            f"Potential high latency sequence started at index {i} with value {value}."
+                        )
+                    consecutive_high_latency_count += 1
+                    if consecutive_high_latency_count > max_spike_width:
+                        # Sustained high latency detected
+                        self.logger.info(
+                            f"Sustained high latency detected, not considered a spike. Sequence starts at index {high_latency_start} and ends at index {i}."
+                        )
+                        return False
+                else:
+                    if 0 < consecutive_high_latency_count <= max_spike_width:
+                        # Isolated spike detected
+                        isolated_spikes.extend(range(high_latency_start, i))
+                        self.logger.info(
+                            f"Isolated spike detected between indices {high_latency_start} and {i-1}."
+                        )
+                    consecutive_high_latency_count = 0
+                    high_latency_start = None
+
+            # Handle case where series ends with a potential spike within allowed width
```

Comment:
sometimes it's easier just to add a "0" to the end of your data than writing a special case like this (as the special case may also need to be updated if the logic above changes, and this is easy to miss)